cli/mender-artifact/validate.go

```diff
@@ -33,18 +33,26 @@ func validate(art io.Reader, key []byte) error {
 	// just continue checking consistency and return info if
 	// signature verification failed
 	var validationError error
+	verifiedDigitalSignature := false
+
+	// Some callers pass nil, others pass "", to indicate a missing public key.
+	isKeySpecified := (key != nil) && (len(key) > 0)
+
 	verify := func(message, sig []byte) error {
 		verifyCallback := func(message, sig []byte) error {
 			return errors.New("artifact is signed but no verification key was provided")
 		}
-		if key != nil {
+
+		if isKeySpecified {
 			s := artifact.NewVerifier(key)
 			verifyCallback = s.Verify
 		}
 
 		if verifyCallback != nil {
 			if err := verifyCallback(message, sig); err != nil {
 				validationError = err
+			} else {
+				verifiedDigitalSignature = true
```

Comment:
Maybe this can be recovered from the reader struct after a read is done? with
`reader.IsSigned` ?